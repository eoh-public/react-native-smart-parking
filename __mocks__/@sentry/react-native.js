export const setUser = jest.fn();
export const init = jest.fn();

export default {
  init,
  setUser,
};
