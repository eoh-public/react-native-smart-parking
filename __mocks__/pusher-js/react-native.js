const channel = {
  bind: jest.fn(),
};

const pusherClient = {
  subscribe: jest.fn(),
  unsubscribe: jest.fn(),
  disconnect: jest.fn(),
};

pusherClient.subscribe.mockImplementation(() => channel);

const MockPusher = jest.fn();
MockPusher.mockImplementation(() => pusherClient);

export default MockPusher;
