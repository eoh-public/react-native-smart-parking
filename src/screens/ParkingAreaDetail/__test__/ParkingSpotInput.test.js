import React, { useState } from 'react';
import { TextInput } from 'react-native';
import { create, act } from 'react-test-renderer';
import ParkingSpotList from '../compenents/ParkingSpotInput';

const mockSetState = jest.fn();
jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn((init) => [init, mockSetState]),
}));

describe('Test ParkingSpotList', () => {
  const mockFunc = jest.fn();
  let wrapper;

  afterEach(() => {
    useState.mockClear();
  });

  test('create ParkingSpotList', async () => {
    useState.mockImplementationOnce((init) => [['2', '4', ''], mockSetState]);
    await act(async () => {
      wrapper = await create(<ParkingSpotList onfinishInputCode={mockFunc} />);
    });

    const instance = wrapper.root;
    const texts = instance.findAllByType(TextInput);
    expect(texts.length).toEqual(3);
    await act(async () => {
      await texts[0].props.onFocus();
      await texts[0].props.onChangeText('2');
      await texts[1].props.onFocus();
      await texts[1].props.onChangeText('1');
      await texts[2].props.onFocus();
      await texts[2].props.onChangeText('2');
    });
    expect(mockSetState).toHaveBeenCalledTimes(6);
    expect(mockFunc).toHaveBeenCalledTimes(0);
  });
  test('render ParkingSpotInput', () => {
    const item = { index: 0, text: { length: 1 } };
    const item1 = { index: 1, text: '' };
    const item2 = { index: 2, text: '' };
    act(() => {
      wrapper = create(<ParkingSpotList onfinishInputCode="function" />);
    });
    const instance = wrapper.root;
    const arrTextInput = instance.findAllByType(TextInput);
    act(() => {
      arrTextInput[0].props.onChangeText({ item });
      arrTextInput[1].props.onChangeText({ item1 });
      arrTextInput[2].props.onChangeText({ item2 });
    });
    expect(arrTextInput).toHaveLength(3);
  });
});
