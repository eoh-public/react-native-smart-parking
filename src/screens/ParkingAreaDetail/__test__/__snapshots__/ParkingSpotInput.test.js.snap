// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ParkingSpotInput create 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "marginTop": 10,
      },
      Object {},
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    maxLength={1}
    name="firstInput"
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`Test ParkingSpotInput create with isFocus 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "marginTop": 10,
      },
      Object {},
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    maxLength={1}
    name="firstInput"
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
